src/Classes/Utils/ColorUtils.ts

```diff
@@ -0,0 +1,199 @@
+            if (color.value instanceof Array) {
+                const arr = color.value as string[];
+                const colorSelected = color.value[Math.floor(Math.random() * arr.length)];
+
+                res = ColorUtils.stringToRgb(colorSelected);
+            } else {
+                const rgbColor = color.value as IRgb;
+
+                if (rgbColor.r !== undefined) {
+                    res = rgbColor;
+                }
+
+                const hslColor = color.value as IHsl;
+
+                if (hslColor.h !== undefined) {
+                    res = ColorUtils.hslToRgb(hslColor);
+                }
+            }
+        }
+
+        return res;
+    }
+
+    public static stringToAlpha(input: string) {
+        return ColorUtils.stringToRgba(input)?.a;
+    }
+
+    /**
+     * Converts hexadecimal string (HTML color code) in a [[IRgb]] object
+     * @param input the hexadecimal string (#f70 or #ff7700)
+     */
+    public static stringToRgb(input: string): IRgb | undefined {
+        return ColorUtils.stringToRgba(input);
+    }
+
+    private static stringToRgba(input: string): IRgba | undefined {
+        if (input.startsWith('rgb')) {
+            const regex = /rgba?\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*(,\s*([\d\.]+)\s*)?\)/i;
+            const result = regex.exec(input);
+
+            return result ? {
+                a: parseInt(result[4]),
+                b: parseInt(result[3]),
+                g: parseInt(result[2]),
+                r: parseInt(result[1]),
+            } : undefined;
+        } else if (input.startsWith('hsl')) {
+            const regex = /hsla?\(\s*(\d+)\s*,\s*(\d+)\%\s*,\s*(\d+)\%\s*(,\s*([\d\.]+)\s*)?\)/i;
+            const result = regex.exec(input);
+
+            return result ? ColorUtils.hslaToRgba({
+                a: parseInt(result[4]),
+                l: parseInt(result[3]),
+                s: parseInt(result[2]),
+                h: parseInt(result[1]),
+            }) : undefined;
+        }
+        else {
+            // By Tim Down - http://stackoverflow.com/a/5624139/3493650
+            // Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
+            const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])([a-f\d])?$/i;
+            const hexFixed = input.replace(shorthandRegex, (_m, r, g, b, a) => {
+                return r + r + g + g + b + b + (a ? a + a : "");
+            });
+            const regex = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})?$/i;
+            const result = regex.exec(hexFixed);
+
+            return result ? {
+                a: parseInt(result[4], 16) / 0xFF,
+                b: parseInt(result[3], 16),
+                g: parseInt(result[2], 16),
+                r: parseInt(result[1], 16),
+            } : undefined;
+        }
+    }
+
+    /**
+     * Converts a Hue Saturation Lightness ([[IHsl]]) object in a [[IRgb]] object
+     * @param hsl
+     */
+    public static hslToRgb(hsl: IHsl): IRgb {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Declaration of public static method not allowed after declaration of private static method. Instead, this should come at the beginning of the class/interface.](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5216054)